extensions/typescript/src/typescriptServiceClient.ts

```diff
@@ -416,65 +496,94 @@ export default class TypeScriptServiceClient implements ITypescriptServiceClient
-			messageShown = window.showInformationMessage<MyMessageItem>(
-				usingWorkspaceVersion
-					? localize(
-						'usingWorkspaceTsVersion',
-						'Using TypeScript version {0} from workspace for IntelliSense.',
-						localVersion)
-					: localize(
-						'usingVSCodeTsVersion',
-						'Using VSCode\'s TypeScript version {0} for IntelliSense.',
-						shippedVersion
-					),
+			messageShown = window.showQuickPick<MyQuickPickItem>([
 				{
-					title: localize('use', 'Use workspace version ({0})', localVersion),
+					label: localize('useWorkspaceVersionOption', 'Use Workspace Version'),
+					description: localVersion || '',
+					detail: modulePath === localModulePath ? localize('activeVersion', 'active') : '',
 					id: MessageAction.useLocal
 				}, {
-					title: localize(
-						'useVSCodeVersionOption',
-						'Use VSCode\'s version ({0})',
-						shippedVersion),
+					label: localize('useVSCodeVersionOption', 'Use VSCode\'s Version'),
+					description: shippedVersion || '',
+					detail: modulePath === this.globalTypescriptPath ? localize('activeVersion', 'active') : '',
 					id: MessageAction.useBundled,
 				}, {
-					title: localize('learnMore', 'Learn More'),
-					id: MessageAction.learnMore,
-					isCloseAffordance: true
+					label: localize('learnMore', 'Learn More'),
+					description: '',
+					id: MessageAction.learnMore
+				}], {
+					placeHolder: usingWorkspaceVersion
+						? localize(
+							'usingWorkspaceTsVersion',
+							'Using TypeScript version {0} from workspace for Typescript language features.',
+							localVersion)
+						: localize(
+							'usingVSCodeTsVersion',
+							'Using VSCode\'s TypeScript version {0} for Typescript language features.',
+							shippedVersion),
 				});
 		} else {
-			messageShown = window.showInformationMessage<MyMessageItem>(
-				localize(
-					'versionCheckUsingBundledTS',
-					'Using VSCode\'s TypeScript version {0} for IntelliSense.',
-					shippedVersion),
+			messageShown = window.showQuickPick<MyQuickPickItem>([
 				{
-					title: localize('learnMore', 'Learn More'),
-					id: MessageAction.learnMore,
-					isCloseAffordance: true,
+					label: localize('learnMore', 'Learn More'),
+					description: '',
+
+					id: MessageAction.learnMore
 				}, {
-					title: localize('close', 'Close'),
-					id: MessageAction.close,
-					isCloseAffordance: true,
+					label: localize('close', 'Close'),
+					description: '',
+					id: MessageAction.close
+				}],
+				{
+					placeHolder: localize(
+						'versionCheckUsingBundledTS',
+						'Using VSCode\'s TypeScript version {0} for Typescript language features.',
+						shippedVersion),
 				});
 		}
 
+		const tryShowRestart = (newModulePath: string) => {
+			if (newModulePath === this.modulePath) {
+				return;
+			}
+
+			window.showInformationMessage<MessageItem>(
+				localize('restartBlurb', 'Restart VSCode to apply change'),
```

Comment:
We don't use `restart` anywhere else. I think we should use `Reload window to apply changes`.